### ISSUE TITLE
LightningDataModule hparam API

### ISSUE THREAD

**Issue by nmichlo** [PyTorchLightning/pytorch-lightning#8051]
EDIT: I am sorry, I somehow completely missed the original proposal: https://github.com/PyTorchLightning/pytorch-lightning/issues/3769

## 🚀 Feature

`hparam` API for `pl.LightningDataModules`

### Motivation

`LightningDataModule`s often have various hyper-parameters passed to them that can affect training, including batch size, dataset split ratios, shuffle and different data transformations, to name a few.

These hyper-parameters should be automatically logged when fitting a model.

### Pitch

It would be great if the same API from `LightningModule`, including`self.hparams`, `self.hparams_initial` and `self.save_hyperparameters()`, was available to `LightningDataModule`.

Additional support in the `pl.Trainer` should be added to automatically log these hyper-parameters when `fit` is called.

### Alternatives

Manually log these hyper parameters.

### Additional context

N/A
